text/0002-numeric-values.md

```diff
@@ -0,0 +1,485 @@
+
+It is ambiguous with a potential `*` unary operator. do you parse `a**b` as:
+
+- `a` `**` `b`
+- `a` `*` `(` `*b` `)`
+
+## Floating point overflow
+
+Should `f64.inf + 1` be `f64.inf` or a crash. This RFC defines that it is `f64.inf`.
+
+# Prior art
+
+[prior-art]: #prior-art
+
+A large part of this RFC is adapted from how Zig's [Integer](https://ziglang.org/documentation/0.11.0/#toc-Integers) and [Float](https://ziglang.org/documentation/0.11.0/#Floats) types work.
+
+The pronounciation of the types is adapted from the C99 [stdint.h](https://en.wikipedia.org/wiki/C_data_types#stdint.h) (`uint32_t`, `int64_t`, etc).
+
+Properties on the type definitions themselves are adapted from Rust's [Integer's Associated Constants](https://doc.rust-lang.org/std/primitive.i32.html)
+
+# Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+## Handling Overflow Errors
+
+A way to handle overflow errors should be given. Making these methods on the values seems like the path of least resistance, but it will lead to long code like `something.tryAdd(63)` or whatever.
+
+Maybe later when we figure out error handling in general we can prescribe a way to handle operator errors easily.
+
+Similarly, this should be the case for `.sqrt`/`.pow` and so on
+
+## Saturating and Wrapping Arithmetic
+
+This proposal does not include saturating or wrapping arithmetic.
+
+Two approaches to consider:
+
+- Rust-like `left.addSaturating(right)`
+- Zig-like `left +| right`
+
+This is a job for a different RFC, as getting the base numberic semantics working is more important.
+
+## Variable naming
+
+Are variable names like `f64` prohibited as an identifier in general? Some languages, including JavaScript, have interesting rules for this.
+
+Valid JavaScript:
+
+```ts
+const x = { class: 2 };
+console.log(x.class);
+```
+
+Valid Zig:
+
+```zig
+pub fn main() void {
+  const x = struct {
+    const @"null" = 1;
+    const @"u8" = 2;
+  };
+
+  const y = x.null;
+  const z = x.u8;
+
+  @import("std").debug.print("{d} {d}", .{y, z});
+}
+```
+
+I'm inclined to allow it, but that isn't neccecarily about this RFC, but about a property access RFC.
+
+## Naming compile-time known integers
+
+This RFC does not name the type of [numeric literals](#type-of-numeric-literals). A pre-requisite to doing this realistically would be adding a full equivilent to `comptime`, which may not be desirable, and is a job for a different RFC.
+
+## Hex/Octal/Binary literals
+
+To keep the scope down, this RFC does not address these literals. Currently, all numeric literals are base 10.
+
+# Future possibilities
+
+[future-possibilities]: #future-possibilities
+
+## More Methods
+
+Later on, it will be cool to add more properties / methods to the integer types, such as `i32.parse(str)`. Exactly how this would compile is not clear at the moment, as `parseInt` is a weird API that we may not want to expose as-is. The integer value can contain methods like `.string()` and others.
```

Comment:
How does this differ from `i32.to(string)`?

Reply:
need to update this. this was in the revision before i theorized `.to(T)`